flake.nix

```diff
@@ -0,0 +1,43 @@
+{
+  description = "Lychee - A fast, async, stream-based link checker";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    fenix.url = "github:nix-community/fenix";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    fenix,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {inherit system;};
```

Comment:
```suggestion
      pkgs = nixpkgs.legacyPackages.${system};
```

Reply:
Importing nixpkgs here is unnecessary overhead.

Reply:
That actually makes a lot of sense, thank you for showing me this